src/utils/hsdis/binutils/hsdis-binutils.c

```diff
@@ -243,10 +243,27 @@ static const char* format_insn_close(const char* close,
   }
 
   strcpy(buf, close);
-  char* p = buf;
-  if (type)    sprintf(p += strlen(p), " type='%s'", type);
-  if (dsize)   sprintf(p += strlen(p), " dsize='%d'", dsize);
-  if (delays)  sprintf(p += strlen(p), " delay='%d'", delays);
+  size_t used_size = strlen(close);
+  char* p = buf + used_size;
+  bufsize -= used_size;
```

Comment:
May not happen in practice, but if `used_size` is larger than `bufsize` this will wrap to a very large value. Perhaps the strcpy above should also be an `snprintf`, and the return value handled the same way as for the subsequent `snprintf` calls?

Reply:
I think it is safe as the `buf` size has been checked at around line 230.  However, it may make the code easier to read if replacing `strcpy` with `snprintf`.  The patch was updated accordingly.

Reply:
This and all uses of snprintf in this change are incorrect.  If the output is truncated, snprintf returns the
number of characters that would have been written if there had been enough space.  That is, the result
may be larger than bufsize.

Reply:
> This and all uses of snprintf in this change are incorrect. If the output is truncated, snprintf returns the number of characters that would have been written if there had been enough space. That is, the result may be larger than bufsize.

The correctness of this change depends on the fact that the buffer has sufficient capacity, which has been checked at line 230.  I agreed that this is not a typical use of `snprintf` that the returned value is not checked.   I will make an update to check the returned value of `snprintf`.

Reply:
OK, I missed that.  (The relevant code doesn't show up in the default github diff.  I really ought to know better
than to use that view for reviewing.)  Even having been pointed to the code, I had to do some counting and
such to convince myself that it was safe.  A bit of commentary might save some time for the next reader.